docs/code/introduction/mode_code.md

```diff
@@ -0,0 +1,126 @@
+
+```
+mode:
+  start_events: ball_starting
+  priority: 100
+  code: base.Base
+```
+
+Note that the value for the `code:` section is the name of the Python module (the file), then a dot, then the name of the class from that file. So in this case, that’s base.Base.
+
+## Run your game!
+
+At this point you should be able to run your game and nothing should happen. This is good, because if it doesn’t crash, that means you did everything right. Of course nothing special happens because you didn’t actually add any code to your custom mode code, so you won’t see anything different.
+
+## Add some custom methods to do things
+
+You can look at the Mode base class (the link from GitHub from earlier) to see what the base Mode class does. However, we have created a few “convenience” methods that you can use. They are:
+
+* **mode_init:** Called once when MPF is starting up
+
+* **mode_start:** Called every time the mode starts, just after the mode_<name>_started event is posted.
+
+* **add_mode_event_handler:** This is the same as the main add_event_handler() method from the Event Manager, except since it’s mode-specific it will also automatically remove any event handlers that you registered when the mode stops. (If you want to register event handlers that are always watching for events even when the mode is not running, you can use the regular self.machine.mode.add_handler() method.
+
+You don’t have to use all of these if you don’t want to.
+
+Also, modes have additional convenience attributes you can use within your mode code:
+
+* **self.config:** A link to the config dictionary for the mode’s config file.
+
+* **self.priority:** The priority the mode is running at. (Don’t change this. Just read it.)
+
+
+* **self.delay:** An instance of the delay manager you can use to set delayed callbacks for this mode. Any active ones will be automatically removed when the mode ends.
+
+* **self.player:** A link to the current player object that’s automatically updated when the player changes. This will be None if the mode is running outside of a game.
+
+* **self.active:** A boolean (True/False) value you can query to see if the mode is running.
+
+## Example usage
+
+Here’s an example of some mode code in use. This example is just a bunch of random things, but again, since you’re writing code here, the sky’s the limit! Seriously you could do all your game logic in mode code and not use the MPF configs at all if you wanted to.
+
+
+```
+from mpf.core.mode import Mode
+
+
+class Base(Mode):
+
+    def mode_init(self):
+        print("My custom mode code is being initialized")
+
+    def mode_start(self, **kwargs):
+        # The mode_start method needs **kwargs because some events that
+        # start modes pass additional parameters
+
+        print("My custom mode code is starting")
+
+        # call a delay in 5 seconds
```

Comment:
`# Set a delay to call self.my_callback() in 5 seconds`

Reply:
changed.